ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkFields/NetworkFields.tsx

```diff
@@ -26,17 +26,50 @@ export type NetworkValues = {
   vlan: NetworkInterface["vlan_id"];
 };
 
+/*
+
+
+ */
+
+/**
+ * Formik values eventually resolve to the correct types,
+ * meanwile we need to force the subnet value to be a number.
+ */
+const toNumber = (value: string | number): number | undefined => {
+  if (typeof value === "number") {
+    return value;
+  }
+  if (isNaN(parseInt(value, 10))) {
+    // Formik requires number fields to be `undefined` when they have no value.
+    return undefined;
+  }
+  return parseInt(value, 10);
+};
+
 const fieldOrder = ["fabric", "vlan", "subnet", "mode", "ip_address"];
 
-const NetworkFields = (): JSX.Element | null => {
+type Props = {
+  editing?: boolean;
+};
+
+const NetworkFields = ({ editing }: Props): JSX.Element | null => {
   const fabrics: Fabric[] = useSelector(fabricSelectors.all);
   const subnets: Subnet[] = useSelector(subnetSelectors.all);
   const { setFieldValue, values } = useFormikContext<NetworkValues>();
   const resetFollowingFields = (name: keyof NetworkValues) => {
     // Reset all fields after this one.
     const position = fieldOrder.indexOf(name);
     for (let i = position + 1; i < fieldOrder.length; i++) {
-      setFieldValue(fieldOrder[i], "");
+      let value: string;
+      switch (fieldOrder[i]) {
```

Comment:
Will this have more values in the future? If not you could initialize `value` as an empty string and then have a simple conditional for re-assigning it depending on the `"mode"`.

Reply:
I suppose we don't expect to add more cases so I've changed this to an `if`.